### DISCUSSION TITLE
Dots-Hyprland by end-4 on Void Linux

### DISCUSSION THREAD

**Discussion by nullptroma** [end-4/dots-hyprland#659] [Extra Distros]
I am currently working on dependency resolution and setting up an environment on Void Linux. I have almost everything ready and soon I will share the script for installing all the dependencies. 

But I ran into a problem: sometimes, I don't know how or why, a second AGS bar appears. Is there an option from which this can happen?

**Comment by nullptroma** [comment]
![image](https://github.com/end-4/dots-hyprland/assets/64524860/923f82bc-0408-46c2-828b-9a2179390730)

**Comment by end-4** [comment]
> I don't know how or why, a second AGS bar appears

An AGS instance registers a bus, by default named "ags". It refuses to run if a bus with such name already exists, so there shouldn't be two running unless you specify a different bus name for each instance.
This only happens on Void right? 
I know in Void there's runit instead of systemd, but I have no clue whether or not that's relevant :thinking:

**Reply by nullptroma** [reply]
It seems that this problem disappeared after the correct sound settings on Void Linux. In any case, this problem appears very rarely now.

**Reply by nullptroma** [reply]
![image](https://github.com/end-4/dots-hyprland/assets/64524860/3a24624d-3f59-45e3-baff-bbbe1c83d45c)

**Comment by nullptroma** [comment]
Hello everybody!
I am attaching a script that will install all the dependencies that the environment requires. Write if something is missing!
Below are my comments on the system setup process, some of them may be funny or obvious, but for me these were important points, since this is the first time I use Void Linux

```bash
#!/bin/bash
set -e
t="$HOME/.cache/depends/"
sudo rm -rf $t
mkdir -p $t
cd $t
sudo xbps-install -S
sudo xbps-install -y gnome-bluetooth bluez-cups bluez
sudo xbps-install -y gtk4-devel libadwaita-devel python3-pip
sudo xbps-install -y coreutils wl-clipboard xdg-utils cmake curl fuzzel rsync wget ripgrep jq nodejs meson gjs axel
sudo npm install --location=global typescript

sudo xbps-install cliphist


# ? libdbusmenu-gtk4
sudo xbps-install -y tinyxml-devel tinyxml tinyxml2 tinyxml2-devel gtkmm-devel gtksourceviewmm-devel gtksourceview5 gtksourceview5-devel gtksourceviewmm-devel gtksourceviewmm cairomm-devel cairomm1.16-devel cairomm cairomm1.16
sudo xbps-install -y python3-build python3-Pillow pywal python3-setuptools_scm python3-wheel
sudo xbps-install -y xrandr xdg-desktop-portal xdg-desktop-portal-gtk xdg-desktop-portal-hyprland
sudo xbps-install -y pavucontrol wireplumber libdbusmenu-gtk3-devel libdbusmenu-gtk3 playerctl swww yad

sudo xbps-install -y scdoc

cd $t
git clone  https://github.com/ReimuNotMoe/ydotool
cd ydotool
mkdir build && cd build
cmake -DSYSTEMD_USER_SERVICE=OFF -DSYSTEMD_SYSTEM_SERVICE=ON ..
make -j `nproc`
sudo cp ydotool /usr/local/bin
sudo cp ydotoold /usr/local/bin
sudo chmod +s $(which ydotool)
sudo mkdir -p /etc/sv/ydotoold
sudo cat << EOF | sudo tee /etc/sv/ydotoold/run
#!/bin/sh
/usr/local/bin/ydotoold
EOF
sudo chmod +x /etc/sv/ydotoold/run
sudo ln -sf /etc/sv/ydotoold /var/service/
sudo sv up ydotoold


sudo xbps-install -y webp-pixbuf-loader gtk-layer-shell-devel wxWidgets-gtk3 wxWidgets-gtk3-devel gtksourceview gtksourceview-devel gobject-introspection upower
sudo xbps-install -y polkit-gnome gnome-keyring gnome-control-center gnome-bluetooth NetworkManager gammastep gnome-bluetooth
sudo xbps-install -y brightnessctl ddcutil

#dart-sass
cd $t
wget https://github.com/sass/dart-sass/releases/download/1.77.0/dart-sass-1.77.0-linux-x64.tar.gz
tar -xzf dart-sass-1.77.0-linux-x64.tar.gz
cd dart-sass
sudo cp -rf * /usr/local/bin/

# ? Mesa-libGLESv2-devel
sudo xbps-install -y python3-pywayland python3-psutil wl-clipboard libwebp-devel file-devel libdrm-devel libgbm-devel pam-devel  libsass-devel libsass


sudo xbps-install -y wlogout


sudo xbps-install -y cargo
cd $t
git clone https://github.com/Kirottu/anyrun.git # Clone the repository
cd anyrun # Change the active directory to it
cargo build --release # Build all the packages
cargo install --path anyrun/ # Install the anyrun binary
sudo cp $HOME/.cargo/bin/anyrun /usr/local/bin/
mkdir -p ~/.config/anyrun/plugins # Create the config directory and the plugins subdirectory
cp target/release/*.so ~/.config/anyrun/plugins # Copy all of the built plugins to the correct directory
cp examples/config.ron ~/.config/anyrun/config.ron # Copy the default config file


# ttf-material-symbols-variable-git ttf-space-mono-nerd ttf-readex-pro jetbrains-mono-fonts gdouros-symbola-fonts lato-fonts
# I do not know how to fix fonts, so I copied them from Arch (all from /usr/share/fonts/ on Arch to /usr/local/share/fonts/ on Void)
sudo xbps-install -y adwaita-plus qt5ct qt6-wayland-devel qt6-wayland-tools qt6-wayland fontconfig fish-shell foot starship
sudo xbps-install -y swappy wf-recorder grim tesseract tesseract-ocr-all slurp
sudo xbps-install -y gobject-introspection gjs-devel pulseaudio pulseaudio-devel python3-devel gettext

# color-generation
# ? libxdp-devel typelib-1_0-Xdp-1_0 typelib-1_0-XdpGtk3-1_0 typelib-1_0-XdpGtk4-1_0
pip3 install materialyoucolor --upgrade --break-system-packages
pip3 install material-color-utilities-python base anyascii svglib libsass --upgrade --break-system-packages
sudo xbps-install -y python3-regex unzip
sudo xbps-install -y python3-gobject-devel libsoup-devel blueprint-compiler xz

sudo xbps-install -y python3-lxml libportal libportal-gtk4 libportal-gtk4-devel bc fcitx5
cd $t
git clone https://github.com/GradienceTeam/Gradience.git
cd Gradience
pip3 install -r requirements.txt --break-system-packages
git submodule init
git submodule update
meson setup builddir
meson configure builddir -Dprefix="$(pwd)/builddir" -Dbuildtype=debug
ninja -C builddir install

echo work!! && exit
```

**Comment by nullptroma** [comment]
1) Gradience. It's a terrible thing. I've spent a lot of time trying to make it work, you should fix its sources yourself so that it comes together. The problems arise due to python version 3.12, it is designed for 3.11. You should also add the variable `echo "PYTHONPATH=/usr/local/lib/python 3.12/site-packages/" >> /etc/environment`, for some reason in Void Linux python initially does not search for modules in this folder. 
2) Cursor theme: https://www.gnome-look.org/p/1914825/, https://wiki.hyprland.org/Hypr-Ecosystem/hyprcursor/
3) Fonts: the same situation as on [openSUSE](https://github.com/end-4/dots-hyprland/discussions/485 ), https://drive.google.com/file/d/14TsjO28Uk837PBhf5bwIj4cLjFEdKXfp/view?usp=sharing
4) Do not forget start `pipewire` with `Hyprland exec-once`

**Comment by nullptroma** [comment]
I am ready to answer any questions about the installation and help if there are difficulties

**Reply by D4vidGarcia** [reply]
So I have this error, I have no idea how to go on this:

```
CMake Error: CMake was unable to find a build program corresponding to "Unix Makefiles".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
```

**Comment by thememika** [comment]
Hi!

I see those instructions are outdated and no longer relevant, since quite a lot has changed.

Recently I have installed Void Linux (musl variant) on a small x86_64 system, since I wanted to try out that distro.
Now I remembered about Hyprland, and found this discussion.
I will be glad to help by creating the updated instructions for it's setup.

From the first sight on available packages in VoidLinux now, I see it will be necessary building GCC 14 from source to compile Hypr-*, since many of the hyprwm projects are using modern C++ features unavailable in GCC 13.
I could try providing .xbps of GCC 14 for the four most popular system variants (musl x86_64, glibc x86_64, musl arm64, glibc arm64), but it would be better if users built GCC themselves (which could be embedded to the install script) and invoked it from a dedicated folder (instead of installing system-wide).
There is a problem that GCC takes quite a while to build on low end systems, but apart from that, I don't see any issues. We could also choose LLVM instead of GCC but that also will need to be built from source.

Regarding python packages, `material-color-utilities-python` fails to install because it depends on an old `Pillow` version which is incompatible with Python 3.13 and fails to install. I don't remember how important is `material-color-utilities-python` to these dotflies because it has been quite a while since I had setup this (I did it in an LFS-like system). I will revisit that one to remember.

There could be more issues to be discovered, and I will notify you about updates.

**Reply by clsty** [reply]
For python packages we now install it via uv into the virtual environment. After that, some code of this project use the env var `$ILLOGICAL_IMPULSE_VIRTUAL_ENV` to get its path to use it. Read the definition of function `install-python-packages` in `scriptdata/installers` for details.

**Reply by Draculagibus** [reply]
> Hi!
> 
> I see those instructions are outdated and no longer relevant, since quite a lot has changed.
> 
> Recently I have installed Void Linux (musl variant) on a small x86_64 system, since I wanted to try out that distro. Now I remembered about Hyprland, and found this discussion. I will be glad to help by creating the updated instructions for it's setup.
> 
> From the first sight on available packages in VoidLinux now, I see it will be necessary building GCC 14 from source to compile Hypr-*, since many of the hyprwm projects are using modern C++ features unavailable in GCC 13. I could try providing .xbps of GCC 14 for the four most popular system variants (musl x86_64, glibc x86_64, musl arm64, glibc arm64), but it would be better if users built GCC themselves (which could be embedded to the install script) and invoked it from a dedicated folder (instead of installing system-wide). There is a problem that GCC takes quite a while to build on low end systems, but apart from that, I don't see any issues. We could also choose LLVM instead of GCC but that also will need to be built from source.
> 
> Regarding python packages, `material-color-utilities-python` fails to install because it depends on an old `Pillow` version which is incompatible with Python 3.13 and fails to install. I don't remember how important is `material-color-utilities-python` to these dotflies because it has been quite a while since I had setup this (I did it in an LFS-like system). I will revisit that one to remember.
> 
> There could be more issues to be discovered, and I will notify you about updates.

Hi !

Do you by any chance kept working on this ?

**Reply by thememika** [reply]
@Draculagibus , sorry, abandoned that because I stopped using Void. Hope it can be easier today if Void now provides gcc 14 from official repo (I'm unsure about it)?

**Reply by rXelelo** [reply]
do you need it? @thememika 
<img width="1680" height="1050" alt="2026-05-14-134210_hyprshot" src="https://github.com/user-attachments/assets/c830018c-a6f0-4e72-863b-734c90124923" />
